### PR TITLE
Cleaning up passing unrequired params in the set_require_fields form

### PR DESCRIPTION
**Pull request by nhoobin**

### CODE REVIEW

File: set_required_fields.php [hunk 1]
```diff
@@ -25,26 +25,44 @@
 
 
 use profilefield_branching\form\set_require_fields_form;
+use profilefield_branching\utility;
 
 require_once(__DIR__.'/../../../../config.php');
 
+// Not the full require_login() check, as it creates a cyclic loop around profilefield_branching_after_require_login().
+if ((!isloggedin() or isguestuser())) {
```
Comment (brendanheywood) [inline set_required_fields.php:40]: I think this is problematic, and probably won't work with saml, or might add extra redirects. We should simply not redirect if we are already on this page which is how other plugins like tool_securityquestions work

File: version.php [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version   = 2015051800;        // The current plugin version (Date: YYYYMMDDXX)
+$plugin->version   = 2021030900;        // The current plugin version (Date: YYYYMMDDXX)
```
Comment (brendanheywood) [inline version.php:27]: Can you please also get github actions working, this can be a new pr if you prefer

Comment (nhoobin) [reply version.php:27]: I'll set that up in a different pull.

File: set_required_fields.php (hunk 1 above)
Comment (nhoobin) [reply set_required_fields.php:40]: I've removed the redirects, and it now prompts a page with a language string, then dies.

**Pull request merged**